### PR TITLE
Passing the success_exit_codes to facts

### PR DESCRIPTION
**Pull request by mohsinhijazee**
Currently, the `success_exit_codes` are not passed down to the facts hence any fact that depends on a command whose only output matters or it has non standard exit codes, cannot be used for fact gathering. This PR fixes that.

### CODE REVIEW

File: pyinfra/api/facts.py [hunk 1]
```diff
@@ -196,6 +197,7 @@ def get_facts(
         ignore_errors = current_global_kwargs.get('ignore_errors', ignore_errors)
         timeout = current_global_kwargs.get('timeout', timeout)
         env = current_global_kwargs.get('env', env)
+        success_exit_codes = current_global_kwargs.get('success_exit_codes', [0])
```
Comment (Fizzadar) [inline pyinfra/api/facts.py]:
```suggestion
        success_exit_codes = current_global_kwargs.get('success_exit_codes', success_exit_codes)
```

**Review by Fizzadar — APPROVED** [review]
Thank you for implementing this @mohsinhijazee! One minor tweak but otherwise looks good to me.

File: pyinfra/api/facts.py (hunk 1 above)
Comment (mohsinhijazee) [reply pyinfra/api/facts.py]: Okay, makes sense actually. Will fix.

**Comment by mohsinhijazee** [comment]
> Thank you for implementing this @mohsinhijazee! One minor tweak but otherwise looks good to me.

Most welcome! I am going to add the tests that would bump the coverage back up. :)

File: tests/test_api/test_api_facts.py [hunk 2]
```diff
@@ -143,6 +146,39 @@ def test_get_fact_error_ignore(self):
             print_input=False,
             print_output=False,
             shell_executable='sh',
+            success_exit_codes=[0],
```
Comment (mohsinhijazee) [inline tests/test_api/test_api_facts.py:149]: Diff is messed up for some reason. I've however shamelessly copied the boilerplate from the existing test case.

File: pyinfra/api/facts.py (hunk 1 above)
Comment (mohsinhijazee) [reply pyinfra/api/facts.py]: Fixed.

**Comment by mohsinhijazee** [comment]
@Fizzadar GTG from my side subject to your approval of course! There's another issue related to the facts that only stdout is returned and I am not sure if we should touch it or not, I'll create an issue for that discussion separately.

**Review by Fizzadar — APPROVED** [review]
Thank you @mohsinhijazee! Looks good to me, will get into `v1.6.3` I'm about to release.

**Pull request merged**